provisioning_client/adapters/hsm_client_riot.c

```diff
@@ -86,25 +86,6 @@ static uint8_t g_CDI[DICE_DIGEST_LENGTH] = {
     0xE9, 0x79, 0xEE, 0xF1, 0x62, 0xF8, 0x64, 0xDA,
     0x50, 0x69, 0x4B, 0x3E, 0x5A, 0x1E, 0x3A, 0x6E };
 
-// The "root" signing key. This is intended for development purposes only.
-// This key is used to sign the DeviceID certificate, the certificiate for
-// this "root" key represents the "trusted" CA for the developer-mode
-// server(s). Again, this is for development purposes only and (obviously)
-// provides no meaningful security whatsoever.
-static unsigned char eccRootPubBytes[sizeof(ecc_publickey)] = {
```

Comment:
Yes, this is really where the MSR team need to chime in. Essentially from my understanding there was 2 ways to produce the root-key, statically & dynamically, I could have done it the way that was described in the email chain statically, that worked perfectly, however that method required inits for 32/64bit OS's and for each of those an init depending on a supported curve ie: 32 bit & 256, 32bit & 384, 32bit & 521, 64bit......

Reply:
@DMattoon, @jebrando, @hihigupt I thought that in all devices the root cert is burned in and fused (my understanding of the RIoT whitepaper). 

@ttins AFAIK the recommended curve is the P256 NIST. I expect the curve itself not to be fully defined but referenced (`secp256r1`).

Reply:
In the reference, the root key is used to sign the DeviceID certificate.  If, for example, I had a bunch of devices I needed to onboard to DPS I would use this root key to sign DeviceID certs of my devices during manufacture.  I would then enroll the cert for that key so when these devices come online the cert chain presented by the device can be validated and will chain to the enrolled root cert.  

In the wild, this root key would never exist outside of a protected location and never on a device.  I think there is an unfortunate overlap in terminology.  This "root" key is a property of the "trusted root CA" not a device's root of trust.

Reply:
This should only relate to creating a cert:
```C
HSM_CLIENT_HANDLE hsm_client_riot_create(void)
{
    ............
    {
        memset(result, 0, sizeof(HSM_CLIENT_X509_INFO));
        if (process_riot_key_info(result) != 0)
     ........
}
```
Otherwise you should be using the other APIs to handle:
```C
char* hsm_client_riot_get_certificate(HSM_CLIENT_HANDLE handle)
char* hsm_client_riot_get_alias_key(HSM_CLIENT_HANDLE handle)
char* hsm_client_riot_get_device_cert(HSM_CLIENT_HANDLE handle)
char* hsm_client_riot_get_signer_cert(HSM_CLIENT_HANDLE handle)
char* hsm_client_riot_get_root_cert(HSM_CLIENT_HANDLE handle)
char* hsm_client_riot_get_root_key(HSM_CLIENT_HANDLE handle)
char* hsm_client_riot_get_common_name(HSM_CLIENT_HANDLE handle)
```
However @jebrando would probably have more insight.

Reply:
We could go back to having it compile time defined, however as stated in the email thread: 

Second (and this is the important one), note the use of RIOT_COORDMAX.  You have the choice of p256, p384, and p521 at build time.  This selection determines the value of RIOT_COORDMAX.  But this function assumes RIOT_COORDMAX == 32 (for p256).  This is fine for a simple example or reference, but “more correct” ways to do this would be to support each curve (with 3 sets of key bytes), or trigger a preprocessor error if this code is compiled with anything other than RIOTSECP256R1 defined.

So this, as stated above would need the key bytes for each curve, and 32/64, to me it seemed like a lot of overhead, where deriving the key uses the RIOT_DIGEST_LENGTH (curve size) to build the key bytes and store them in the x509_info_tag, I will leave this one open to best practice suggestion

Reply:
@DMattoon , @ttins I now understand the reasons for the root key. In Provisioning Service this is called a "group": the root needs to be uploaded to the service and verified (a new leaf cert signed by the root must be produced with a random name that the service imposes).
Because of this, we need the certificate to be fixed so that customers can use the sample. Generating would change the certificate every time which requires the complicated process of validating the certificate.